subprojects/core/src/integTest/groovy/org/gradle/api/internal/tasks/TaskCacheabilityReasonIntegrationTest.groovy

```diff
@@ -133,20 +133,22 @@ class TaskCacheabilityReasonIntegrationTest extends AbstractIntegrationSpec impl
         withBuildCache().run "cacheable"
     }
 
-    def "cacheability for a task with plural outputs is PLURAL_OUTPUTS"() {
+    def "cacheability for a task with file tree outputs is PLURAL_OUTPUTS"() {
         buildFile << """
             @CacheableTask
             class PluralOutputs extends BaseTask {
                 @OutputFiles
-                Set<File> outputFiles = ['some.txt']
+                FileTree outputFiles = project.fileTree('build/some-dir') {
+                    exclude "*.log"
+                }
                 
                 @TaskAction
                 void generate() {}
             }
             
             task cacheable(type: PluralOutputs) {
                 cachingEnabled = false
-                disabledReason = "Declares multiple output files for the single output property 'outputFiles' via `@OutputFiles`, `@OutputDirectories` or `TaskOutputs.files()`"
+                disabledReason = "Declares complex output for the output property 'outputFiles'"
```

Comment:
That will probably require an update in build scans as well. Currently, the reason is "Multiple outputs declared": 
![image](https://user-images.githubusercontent.com/423186/44711330-67f5e600-aaae-11e8-80ea-e5938c329691.png)

I didn't find any help text for it, I guess we would need to add a section to the user guide or the build cache guide explaining what kind of outputs we support for caching and link to it. WDYT?

Reply:
I also think that `complex` is very vague. Can we be more specific in what is allowed? Something like "Only files, directories and unfiltered file trees are allowed"?